### DISCUSSION TITLE
onSelect of CommandItem gives label instead of value of an option. How to get the value instead??

### DISCUSSION THREAD

**Discussion by subhranshudas** [shadcn-ui/ui#863] [Q&A] [answered]
i am looping over a `domainList` and using the Combobox. Below is the part for the `CommandGroup`. this snippet is basically a modification of https://ui.shadcn.com/docs/components/combobox#combobox

```
  <CommandGroup>
      {domainList.map((domain) => (
        <CommandItem
          key={domain.value}
            onSelect={(currentLabel) => { // --> this place
            setValue(currentLabel === value ? "" : currentLabel)
            setOpen(false)
          }}
        >
          <Check
            className={cn(
              "mr-2 h-4 w-4",
              value === domain.value ? "opacity-100" : "opacity-0"
            )}
          />
          {domain.label}
        </CommandItem>
      ))}
    </CommandGroup>
```

What i observed was the `onSelect` is basically giving access to the label value or the text content, but not the `domain.value`. where `domain = { label: "", value: "" }`

Can we get the `value` instead of the label in the **onSelect** method?? (for obvious reasons!)

**Comment by joaom00** [comment, marked as answer]
You should pass `domain.value` in `value` prop

**Reply by subhranshudas** [reply]
Yes Sir, this works!!
